File: Codes in Python3/Binary Tree/Maximum Path Sum in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def helper(self, root,maxi:List[int]):
        if root==None:
            return 0
        leftsum=max(0,self.helper(root.left,maxi))
        rightsum=max(0,self.helper(root.right,maxi))
        maxi[0]=max(maxi[0],root.val+leftsum+rightsum)
        return root.val+max(leftsum,rightsum)

    def maxPathSum(self, root: Optional[TreeNode],) -> int:
        maxi=[float('-inf')]
        self.helper(root,maxi)
        return maxi[0]

File: Codes in C++/Binary Tree/Maximum Path Sum in Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int helper(TreeNode* root,int &maxi)
    {
        if(root==NULL)
            return 0;
        int leftsum=max(0,helper(root->left,maxi));
        int rightsum=max(0,helper(root->right,maxi));
        maxi=max(maxi,root->val+leftsum+rightsum);    
        return root->val+max(leftsum,rightsum);    
    }
    int maxPathSum(TreeNode* root) {
        int maxi=INT_MIN;
        helper(root,maxi);
        return maxi;
    }
};